torchvision/ops/boxes.py

```diff
@@ -398,7 +398,40 @@ def _box_diou_iou(boxes1: Tensor, boxes2: Tensor, eps: float = 1e-7) -> Tuple[Te
     # distance between boxes' centers squared.
     return iou - (centers_distance_squared / diagonal_distance_squared), iou
 
+def masks_to_boundaries(masks: torch.Tensor, dilation_ratio: float = 0.02) -> torch.Tensor:
+    """
+    Compute the boundaries around the provided masks using morphological operations.
+
+    Returns a tensor of the same shape as the input masks containing the boundaries of each mask.
+
+    Args:
+    masks (Tensor[N, H, W]): masks to transform where N is the number of masks
+    and (H, W) are the spatial dimensions.
+    dilation_ratio (float, optional): ratio used for the dilation operation. Default: 0.02
+
+    Returns:
+    Tensor[N, H, W]: boundaries
+    """
+    # If no masks are provided, return an empty tensor
+    if masks.numel() == 0:
+        return torch.zeros_like(masks)
+
+    n, h, w = masks.shape
+    img_diag = math.sqrt(h ** 2 + w ** 2)
+    dilation = int(round(dilation_ratio * img_diag))
+    selem_size = dilation * 2 + 1
```

Comment:
I think it is incorrect to define dilation size by the input size. Same by size masks wont have the same boundaries if canvas are different.
Here is an example with canvas size 32, 120 and 320:

Canvas Size: 32 | Canvas Size: 120 | Canvas Size: 320
---|---|---
![image](https://github.com/pytorch/vision/assets/2459423/f365ed1b-4b80-46bd-a267-87eaefed1d33) | ![image](https://github.com/pytorch/vision/assets/2459423/94efb301-17af-41a7-8752-9262d7aa6a72) | ![image](https://github.com/pytorch/vision/assets/2459423/a1ac6a8f-1f5e-4a15-b998-d71051cd9f24)

All plots are showing the top-left crop of size 32x32 


Another problem with the implementation is that it outputs contour as 4D tensor, for example: `torch.Size([4, 4, 32, 32])` but docs claims to return 3D tensor.

Reply:
It is in the original reference impl of the paper that is related to the image diag. 
See:
https://github.com/bowenc0221/boundary-iou-api/blob/master/boundary_iou%2Futils%2Fboundary_utils.py#L21

Reply:
I do not think that this ref implementation is correct due to above results. They may have assumptions like using masks of the same size. IMO, if we provide mask to boundaries functionality in torchvision, it should be a generic function without any specific reference like above. I wont be able to approve the current state of the PR which aims another goal.

Reply:
No problem but after 8 months of no signal I think  you could have a more collaborative approach in the case you want to attract developers.

So if you want to be constructive we could agree on an alternative implementation without strictly reproducing BoundaryIOU paper.

What do you want to have?

Reply:
As I said in the previous messages we need a generic implementation for mask_to_boundary without replicating BoundaryIOU paper. It should at least fix above issues with different canvas sizes. I would put dillation kernel size as input arg instead of `dilation_ratio`.

Reply:
Ok we will  let the user specify the kernel size. What param name do you want?

Reply:
whatever suites to the function and its purpose

Reply:
Ok you can make another pass on the new refactoring.